test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/ValueFormTests/KebabCaseValueFormTests.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.ValueForms;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests.ValueFormTests
+{
+    public class KebabCaseValueFormTests
+    {
+        [Theory]
+        [InlineData("I", "i")]
+        [InlineData("IO", "io")]
+        [InlineData("FileIO", "file-io")]
+        [InlineData("SignalR", "signal-r")]
+        [InlineData("IOStream", "io-stream")]
+        [InlineData("COMObject", "com-object")]
+        [InlineData("WebAPI", "web-api")]
+        [InlineData("XProjectX", "x-project-x")]
+        [InlineData("NextXXXProject", "next-xxx-project")]
+        [InlineData("NoNewProject", "no-new-project")]
+        [InlineData("NONewProject", "no-new-project")]
+        [InlineData("NewProjectName", "new-project-name")]
+        [InlineData("ABBREVIATIONAndSomeName", "abbreviation-and-some-name")]
+        [InlineData("NoNoNoNoNoNoNoName", "no-no-no-no-no-no-no-name")]
+        [InlineData("AnotherNewNewNewNewNewProjectName", "another-new-new-new-new-new-project-name")]
+        [InlineData("Param1TestValue", "param-1-test-value")]
```

Comment:
Consider including the case when we have 2 or more consecutive digits in the string. (https://stackoverflow.com/a/54012346/164680 gives a good set of use cases, I suppose).